src/types.rs

```diff
@@ -616,7 +616,15 @@ impl Rewrite for ast::TraitRef {
 impl Rewrite for ast::Ty {
     fn rewrite(&self, context: &RewriteContext, shape: Shape) -> Option<String> {
         match self.node {
-            ast::TyKind::TraitObject(ref bounds, ..) => bounds.rewrite(context, shape),
+            ast::TyKind::TraitObject(ref bounds, tobj_syntax) => {
+                let res = bounds.rewrite(context, shape)?;
```

Comment:
Could you please add `.offset_left(4)` when `dyn` is used to make sure that we do not exceed max width with a long trait? For example, we need something like the following:

```rust
let shape = if tobj_syntax == ast::TraitObjectSyntax::Dyn {
    // 4 = "dyn "
    shape.offset_left(4)?
} else {
    shape
};
```

And it would be nice if there is a test for this, though I cannot come up with a good example.

Reply:
I added, but it doesn't seem enough.
When trying to format this,
```rust
fn f3() -> Box<Very__________________________Long__________________Name____________________Trait>
{

}
```
rustfmt cause an error, `error: line exceeded maximum width (maximum: 100, found: 101)`.
However, in this case,
```rust
fn f3() -> Box<dyn Very__________________________Long__________________Name____________________Trait>
{

}
```
rustfmt outputs 
```rust
fn f3(
) -> Box<dyn Very__________________________Long__________________Name____________________Trait>
{

}
```

Reply:
I checked rustfmt formats 
```rust
fn main() {
    fn f3() -> Box<Very______________________________Long__________________Name____________________Trait>
    {
    }
}
```
into 
```rust
fn main() {
    fn f3(
) -> Box<Very______________________________Long__________________Name____________________Trait>
    {
    }
}
```

Though I'm not sure it's a bug, but I think hard to test line exceeding, because of this.